system/lib/wasmfs/wasmfs.cpp

```diff
@@ -9,25 +9,35 @@
 #include <emscripten/emscripten.h>
 #include <emscripten/html5.h>
 #include <stdlib.h>
+#include <vector>
 #include <wasi/api.h>
 
 extern "C" {
 
-int emscripten_wasmfs_printbuffer(__wasi_fd_t fd, const uint8_t* ptr, __wasi_size_t len);
+std::vector<char> buffer;
 
 __wasi_errno_t __wasi_fd_write(
   __wasi_fd_t fd, const __wasi_ciovec_t* iovs, size_t iovs_len, __wasi_size_t* nwritten) {
   // FD 1 = STDOUT and FD 2 = STDERR.
   // Temporary hardcoding of filedescriptor values.
   // TODO: May not want to proxy stderr (fd == 2) to the main thread.
-  // This will not not show in HTML, a console.warn in a worker is suffficient.
+  // This will not show in HTML - a console.warn in a worker is suffficient.
   // This would be a change from the current FS.
   if (fd == 1 || fd == 2) {
     __wasi_size_t num = 0;
     for (size_t i = 0; i < iovs_len; i++) {
       const uint8_t* ptr = iovs[i].buf;
```

Comment:
I would probably call this `buf` rather than `ptr` and use `buf[j]` rather than `*(buf + j)` down below. Also, it would be good to use `uint8_t` everywhere rather than mixing it with `char`.